nixos/modules/services/web-apps/engelsystem.nix

```diff
@@ -0,0 +1,159 @@
+{ config, lib, pkgs, utils, ... }:
+
+let
+  inherit (lib) mkDefault mkEnableOption mkIf mkOption types literalExample;
+
+  cfg = config.services.engelsystem;
+  fpm = config.services.phpfpm.pools.engelsystem;
+
+  genConfigScript = pkgs.writeScript "engelsystem-gen-config.sh" (utils.genJqSecretsReplacementSnippet cfg.config "config.json");
+in {
+  options = {
+    services.engelsystem = {
+      enable = mkEnableOption "Online tool for coordinating helpers and shifts on large events";
+
+      domain = mkOption {
+        type = types.str;
+        example = "engelsystem.example.com";
+        description = "domain to serve on";
+      };
+
+      package = mkOption {
+        type = types.package;
+        example = literalExample "pkgs.engelsystem";
+        description = "Engelsystem package used for the service";
+        default = pkgs.engelsystem;
+      };
+
+      createDatabase = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether to create a local database automatically.
+          This will overrid every database setting in <option>services.engelsystem.config</option>
+        '';
+      };
+    };
+
+    services.engelsystem.config = mkOption {
+      type = types.attrs;
+      default = {
+        database = {
+          unix_socket = "/var/run/mysqld/mysqld.sock";
+          host = "localhost";
+          database = "engelsystem";
+          username = "engelsystem";
+          password = "";
+        };
+      };
+      description = "Options in config.php";
```

Comment:
The description should be expanded. Maybe you could link to https://engelsystem.de/doc/admin/configuration/ and add some information about the secret parameter functionality with an example? See the description and example of `services.gitlab.extraConfig` for reference.